crates/core/src/types/account.rs

```diff
@@ -2,17 +2,53 @@ use std::collections::HashMap;
 
 use bytes::Bytes;
 use ethereum_types::{H256, U256};
-use serde::Deserialize;
+use tiny_keccak::{Hasher, Sha3};
+
+use crate::rlp::encode::RLPEncode;
+
+use super::GenesisAccount;
 
 #[allow(unused)]
-#[derive(Debug, Deserialize, PartialEq)]
+#[derive(Debug, PartialEq)]
 pub struct Account {
-    #[serde(default)]
+    pub info: AccountInfo,
     pub code: Bytes,
-    #[serde(default)]
     pub storage: HashMap<H256, H256>,
-    #[serde(deserialize_with = "crate::serde_utils::u256::deser_dec_str")]
+}
+
+#[derive(Debug, PartialEq)]
+pub struct AccountInfo {
+    pub code_hash: H256,
     pub balance: U256,
-    #[serde(default, deserialize_with = "crate::serde_utils::u64::deser_dec_str")]
     pub nonce: u64,
 }
+
+impl From<GenesisAccount> for Account {
+    fn from(genesis: GenesisAccount) -> Self {
+        Self {
+            info: AccountInfo {
+                code_hash: code_hash(&genesis.code),
+                balance: genesis.balance,
+                nonce: genesis.nonce,
+            },
+            code: genesis.code,
+            storage: genesis.storage,
+        }
+    }
+}
+
+fn code_hash(code: &Bytes) -> H256 {
+    let mut sha3 = Sha3::v256();
+    let mut output = [0; 32];
+    sha3.update(code.as_ref());
+    sha3.finalize(&mut output);
```

Comment:
Can you check that an empty hash (init + finalize) results in `c5d2460186f7233c927e7db2dcc703c0e500b653ca82273b7bfad8045d85a470`? I think this is [a different version of keccak](https://ethereum.stackexchange.com/questions/550/which-cryptographic-hash-function-does-ethereum-use).

Reply:
It yields a different result, Ill look for a compatible crate

Reply:
Thanks for pointing it out!

Reply:
Fixed